src/Message/AbstractRequest.php

```diff
@@ -154,21 +177,24 @@ public function setSource($value)
      */
     protected function getCardData()
     {
-        $this->getCard()->validate();
+        $card = $this->getCard();
+        $card->validate();
 
         $data = array();
         $data['object'] = 'card';
-        $data['number'] = $this->getCard()->getNumber();
-        $data['exp_month'] = $this->getCard()->getExpiryMonth();
-        $data['exp_year'] = $this->getCard()->getExpiryYear();
-        $data['cvc'] = $this->getCard()->getCvv();
-        $data['name'] = $this->getCard()->getName();
-        $data['address_line1'] = $this->getCard()->getAddress1();
-        $data['address_line2'] = $this->getCard()->getAddress2();
-        $data['address_city'] = $this->getCard()->getCity();
-        $data['address_zip'] = $this->getCard()->getPostcode();
-        $data['address_state'] = $this->getCard()->getState();
-        $data['address_country'] = $this->getCard()->getCountry();
+        $data['number'] = $card->getNumber();
+        $data['exp_month'] = $card->getExpiryMonth();
+        $data['exp_year'] = $card->getExpiryYear();
+        if ($this->getCard()->getCvv()) {
```

Comment:
Yes, you're correct, laziness from copying someone else's code.  I will fix this shortly.